### PR TITLE
Task completed: GP-127. Controls: Keyboard

### PR DESCRIPTION
**Pull request by frnzee**
- added "new" Unity InputSystem package
- built input actions map
- added controls manager and debug.log entities for clarity it's working

Jira task: https://lighteatersstudio.atlassian.net/browse/GP-127?atlOrigin=eyJpIjoiZjI3N2M2ZTc4ZDk0NDkzNWJiN2UzMjk5YmU4ZWVmZGUiLCJwIjoiaiJ9

### CODE REVIEW

File: Assets/Prefabs/Gameplay/PlayerInputActionsMap.inputactions
Comment (ega-ega) [inline Assets/Prefabs/Gameplay/PlayerInputActionsMap.inputactions:1]: Move this file to folder: Gameplay/Input

File: Assets/Scripts/Services/Input/InputController.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        [Inject]
+        public InputController(InputActionAsset inputActionAsset)
+        {
+            _inputActionAsset = inputActionAsset;
+            OnEnable();
+        }
+
+        private void Move(InputAction.CallbackContext ctx)
+        {
+            MoveChanged?.Invoke(Vector2.zero);
```
Comment (ega-ega) [inline Assets/Scripts/Services/Input/InputController.cs]: Pls change zero vector to specific value from ctx parametr

File: Assets/Scripts/Services/Input/InputController.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        }
+
+        private void Fire(InputAction.CallbackContext ctx)
+        {
+            FireChanged?.Invoke(Vector2.zero);
```
Comment (ega-ega) [inline Assets/Scripts/Services/Input/InputController.cs]: Pls change zero vector to specific value from ctx parametr

File: Assets/Scripts/Services/Input/InputController.cs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        }
+
+        private void Special(InputAction.CallbackContext ctx)
+        {
+            SpecialChanged?.Invoke(Vector2.zero);
```
Comment (ega-ega) [inline Assets/Scripts/Services/Input/InputController.cs]: Pls change zero vector to specific value from ctx parametr

File: Assets/Scripts/Services/Input/InputController.cs [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        public event Action ReloadChanged;
+
+        [Inject]
+        public InputController(InputActionAsset inputActionAsset)
+        {
+            _inputActionAsset = inputActionAsset;
+            OnEnable();
+        }
+
+        private void Move(InputAction.CallbackContext ctx)
```
Comment (ega-ega) [inline Assets/Scripts/Services/Input/InputController.cs]: Pls rename parametr to context. By our agreement, the use of abbreviations or acronyms must not be allowed. as an exception to this rule, we can take the name "UI" or "FX", which is currently recognized as a concept and not an acronym by the vast majority of people / programmers

File: Assets/Scripts/Services/Input/InputController.cs [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        {
+            MoveChanged?.Invoke(Vector2.zero);
+        }
+
+        private void Fire(InputAction.CallbackContext ctx)
```
Comment (ega-ega) [inline Assets/Scripts/Services/Input/InputController.cs]: Pls rename parametr to context. By our agreement, the use of abbreviations or acronyms must not be allowed. as an exception to this rule, we can take the name "UI" or "FX", which is currently recognized as a concept and not an acronym by the vast majority of people / programmers

File: Assets/Scripts/Services/Input/InputController.cs [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+        {
+            FireChanged?.Invoke(Vector2.zero);
+        }
+
+        private void Special(InputAction.CallbackContext ctx)
```
Comment (ega-ega) [inline Assets/Scripts/Services/Input/InputController.cs]: Pls rename parametr to context. By our agreement, the use of abbreviations or acronyms must not be allowed. as an exception to this rule, we can take the name "UI" or "FX", which is currently recognized as a concept and not an acronym by the vast majority of people / programmers

File: Assets/Scripts/Services/Input/InputController.cs [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+        {
+            SpecialChanged?.Invoke(Vector2.zero);
+        }
+
+        private void Melee(InputAction.CallbackContext ctx)
```
Comment (ega-ega) [inline Assets/Scripts/Services/Input/InputController.cs]: Pls rename parametr to context. By our agreement, the use of abbreviations or acronyms must not be allowed. as an exception to this rule, we can take the name "UI" or "FX", which is currently recognized as a concept and not an acronym by the vast majority of people / programmers

File: Assets/Scripts/Services/Input/InputController.cs [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+        {
+            MeleeChanged?.Invoke();
+        }
+
+        private void Use(InputAction.CallbackContext ctx)
```
Comment (ega-ega) [inline Assets/Scripts/Services/Input/InputController.cs]: Pls rename parametr to context. By our agreement, the use of abbreviations or acronyms must not be allowed. as an exception to this rule, we can take the name "UI" or "FX", which is currently recognized as a concept and not an acronym by the vast majority of people / programmers

File: Assets/Scripts/Services/Input/InputController.cs [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+        {
+            UseChanged?.Invoke();
+        }
+
+        private void Reload(InputAction.CallbackContext ctx)
```
Comment (ega-ega) [inline Assets/Scripts/Services/Input/InputController.cs]: Pls rename parametr to context. By our agreement, the use of abbreviations or acronyms must not be allowed. as an exception to this rule, we can take the name "UI" or "FX", which is currently recognized as a concept and not an acronym by the vast majority of people / programmers

File: Assets/Scripts/Services/Input/InputController.cs [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+        {
+            ReloadChanged?.Invoke();
+        }
+
+        private void OnEnable()
+        {
+            _inputActionAsset.FindAction("Move").performed += Move;
```
Comment (ega-ega) [inline Assets/Scripts/Services/Input/InputController.cs]: Pls extract string value to const.By our convention, your implementation falls under the magic number/value rule.
Also select the rest of the strings as constants

File: Assets/Scripts/Services/Input/InputController.cs (hunk 4 above)
Comment (ega-ega) [inline Assets/Scripts/Services/Input/InputController.cs]: Rename Method to OnMove. Also pls rename all another handlers in this class

**Review by ega-ega — CHANGES_REQUESTED** [review]

File: Assets/Prefabs/Gameplay/PlayerInputActionsMap.inputactions
Comment (frnzee) [reply Assets/Prefabs/Gameplay/PlayerInputActionsMap.inputactions:1]: Done

File: Assets/Scripts/Services/Input/InputController.cs (hunk 1 above)
Comment (frnzee) [reply Assets/Scripts/Services/Input/InputController.cs]: Done

File: Assets/Scripts/Services/Input/InputController.cs (hunk 2 above)
Comment (frnzee) [reply Assets/Scripts/Services/Input/InputController.cs]: Done

File: Assets/Scripts/Services/Input/InputController.cs (hunk 3 above)
Comment (frnzee) [reply Assets/Scripts/Services/Input/InputController.cs]: Done

File: Assets/Scripts/Services/Input/InputController.cs (hunk 4 above)
Comment (frnzee) [reply Assets/Scripts/Services/Input/InputController.cs]: Done

File: Assets/Scripts/Services/Input/InputController.cs (hunk 7 above)
Comment (frnzee) [reply Assets/Scripts/Services/Input/InputController.cs]: Done

File: Assets/Scripts/Services/Input/InputController.cs (hunk 8 above)
Comment (frnzee) [reply Assets/Scripts/Services/Input/InputController.cs]: Done

File: Assets/Scripts/Services/Input/InputController.cs (hunk 9 above)
Comment (frnzee) [reply Assets/Scripts/Services/Input/InputController.cs]: Done

File: Assets/Scripts/Services/Input/InputController.cs (hunk 10 above)
Comment (frnzee) [reply Assets/Scripts/Services/Input/InputController.cs]: Done

File: Assets/Scripts/Services/Input/InputController.cs (hunk 4 above)
Comment (frnzee) [reply Assets/Scripts/Services/Input/InputController.cs]: Done

File: Assets/Scripts/Services/Input/InputController.cs (hunk 5 above)
Comment (frnzee) [reply Assets/Scripts/Services/Input/InputController.cs]: Done

File: Assets/Scripts/Services/Input/InputController.cs (hunk 6 above)
Comment (frnzee) [reply Assets/Scripts/Services/Input/InputController.cs]: Done

File: Assets/Scripts/Installer/GameInstaller.cs [hunk 11]
```diff
@@ -24,8 +26,11 @@ public class GameInstaller : MonoInstaller
         [SerializeField] private Player _playerPrefab;
         [SerializeField] private Character _characterPrefab;
         [SerializeField] private HealthBar _healthBarPrefab;
-        
-        
+
+        [Header("Input")]
+        [SerializeField] private InputActionAsset _playerInputActionsMap;
+        private InputController _playerInputController;
```
Comment (ega-ega) [inline Assets/Scripts/Installer/GameInstaller.cs]: Pls remove unused field

**Review by ega-ega — CHANGES_REQUESTED** [review]

**Review by ega-ega — APPROVED** [review]

**Review by Kjusok — APPROVED** [review]

**Pull request merged**